packages/sources/cfbenchmarks-test/src/endpoint/rest/crypto.ts

```diff
@@ -0,0 +1,74 @@
+import { BatchWarmingTransport } from '@chainlink/external-adapter-framework/transports'
+import { EmptyObject } from '@chainlink/external-adapter-framework/util'
+import { EndpointTypes, getIdFromBaseQuote } from '../common/crypto'
+
+interface PayloadValue {
+  value: string
+  time: number
+}
+
+interface ProviderResponse {
+  payload: PayloadValue[]
+}
+
+export type RestEndpointTypes = EndpointTypes & {
+  Provider: {
+    RequestBody: EmptyObject
+    ResponseBody: ProviderResponse
+  }
+}
+
+// NOTE: This is using the BatchWarming transport, but the actual API endpoint is not batchable
+export const transport = new BatchWarmingTransport<RestEndpointTypes>({
+  prepareRequest: ([{ base, quote }], { DEFAULT_API_ENDPOINT }) => {
+    return {
+      baseURL: DEFAULT_API_ENDPOINT,
+      url: '/v1/values',
+      method: 'GET',
+      params: {
+        id: getIdFromBaseQuote(base, quote, 'primary'),
```

Comment:
This is failing in a bunch of CI runs 🤔 
`Argument of type 'string | undefined' is not assignable to parameter of type 'string'.`

Reply:
Damn, from the required input parameter groups. Coercing them and adding a note.